contracts/staking/contracts/src/stake/MixinStake.sol

```diff
@@ -17,188 +17,217 @@
-/// 3. Deactivated & TimeLocked
-/// 4. Deactivated & Withdrawable
-///
-/// -- State Definitions --
-/// Activated Stake
-///     Stake in this state can be used as a utility within the 0x ecosystem.
-///     For example, it carries weight when computing fee-based rewards (see MixinExchangeFees).
-///     In the future, it may be used to participate in the 0x governance system.
-///
-/// Activated & Delegated Stake
-///     Stake in this state also serves as a utility that is shared between the delegator and delegate.
-///     For example, if delegated to a staking pool then it carries weight when computing fee-based rewards for
-///     the staking pool; however, in this case, delegated stake carries less weight that regular stake (see MixinStakingPool).
-///
-/// Deactivated & TimeLocked Stake
-///     Stake in this state cannot be used as a utility within the 0x ecosystem.
-///     Stake is timeLocked when it moves out of activated states (Activated / Activated & Delagated).
-///     By limiting the portability of stake, we mitigate undesirable behavior such as switching staking pools
-///     in the middle of an epoch.
-///
-/// Deactivated & Withdrawable
-///     Stake in this state cannot be used as a utility with in the 0x ecosystem.
-///     This stake can, however, be burned and withdrawn as Zrx tokens.
-/// ----------------------------
-///
-/// -- Valid State Transtions --
-/// Activated -> Deactivated & TimeLocked
-///
-/// Activated & Delegated -> Deactivated & TimeLocked
-///
-/// Deactivated & TimeLocked -> Deactivated & Withdrawable
-///
-/// Deactivated & Withdrawable -> Activated
-/// Deactivated & Withdrawable -> Activated & Delegated
-/// Deactivated & Withdrawable -> Deactivated & Withdrawable
-/// ----------------------------
-///
-/// Freshly minted stake is in the "Deactvated & Withdrawable" State, so it can
-/// either be activated, delegated or withdrawn.
-/// See MixinDelegatedStake and MixinTimeLockedStake for more on respective state transitions.
 contract MixinStake is
     IStakingEvents,
     MixinDeploymentConstants,
+    Ownable,
     MixinConstants,
     MixinStorage,
+    MixinZrxVault,
     MixinScheduler,
     MixinStakingPoolRewardVault,
-    MixinZrxVault,
-    MixinTimeLockedStake,
-    MixinStakeBalances
+    MixinStakeStorage,
+    MixinStakeBalances,
+    MixinStakingPoolRewards
 {
     using LibSafeMath for uint256;
 
-    /// @dev Deposit Zrx. This mints stake for the sender that is in the "Deactivated & Withdrawable" state.
-    /// @param amount of Zrx to deposit / Stake to mint.
-    function depositZrxAndMintDeactivatedStake(uint256 amount)
+    /// @dev Stake ZRX tokens. Tokens are deposited into the ZRX Vault. Unstake to retrieve the ZRX.
+    ///      Stake is in the 'Active' state.
+    /// @param amount of ZRX to stake.
+    function stake(uint256 amount)
         external
     {
-        _mintStake(msg.sender, amount);
+        address payable owner = msg.sender;
```

Comment:
Hmm I hadn't thought of that. How about `staker`?

Reply:
Yeah we're going to need to change this in a lot of places. This name is used all throughout the codebase. I've created a task to do this once all outstanding PR's are merged, to avoid all the conflicts.